libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorRecognizer.cs

```diff
@@ -417,7 +425,37 @@ private void InitializeModel()
             byte[] snapShotByteArray = Encoding.UTF8.GetBytes(content);
 
             // Create label resolver
-            _resolver = orchestrator.CreateLabelResolver(snapShotByteArray);
+            _resolver = _orchestratorDictionaryEntry.Orchestrator.CreateLabelResolver(snapShotByteArray);
+        }
+
+        /// <summary>
+        /// OrchestratorDictionaryEntry is used in a static OrchestratorMap object.
+        /// </summary>
+        protected class OrchestratorDictionaryEntry
```

Comment:
Is this class used outside of this assembly? If not, please make it private. Protected members get exported publicly and we need to maintain backward compat for the foreseeable future.

Reply:
It would be tricky, but if I inherit from OrchestratorRecognizer as a bot developer, I'd be able to access this class and us such, this becomes part of our public surface and needs to be treated as such. From what I observe, there is no value in making that commitment at this point. Could we mark it as private class instead?

Reply:
Example:

Assembly A:

public class Outer { protected class Inner {}}

Assembly B (references A):

public class Foo {  public Foo() { var inner = new Inner();}} // this works!

Reply:
Will make the class private.